java-checks/src/main/java/org/sonar/java/checks/SAMAnnotatedCheck.java

```diff
@@ -66,6 +70,11 @@
       build();
 
   @Override
+  public boolean isCompatibleWithJavaVersion(Integer version) {
+    return JavaVersionHelper.java8Compatible(version);
```

Comment:
Indeed if the rule is enabled by default, having the adequate version should be guaranteed.